### ISSUE TITLE
manticore should terminate state on suicide

### ISSUE THREAD

**Issue by mossberg** [trailofbits/manticore#721]
```
contract XX {

    function sui() {
        suicide(msg.sender);
    }

}
```

manticore seemingly keeps the state alive, even though it killed itself. we should generate a testcase for this?

```
2018-01-31 20:07:39,946: [116141] m.main:INFO: Beginning analysis
2018-01-31 20:07:40,817: [116141] m.ethereum:INFO: Starting symbolic transaction: 1
2018-01-31 20:07:42,159: [116141] m.ethereum:INFO: Generated testcase No. 0 - REVERT
2018-01-31 20:07:45,508: [116141] m.ethereum:INFO: Generated testcase No. 1 - REVERT
2018-01-31 20:07:48,182: [116141] m.ethereum:INFO: Generated testcase No. 2 - REVERT
2018-01-31 20:07:51,988: [116141] m.ethereum:INFO: Generated testcase No. 3 - REVERT
2018-01-31 20:07:54,484: [116141] m.ethereum:INFO: Finished symbolic transaction: 1 | Code Coverage: 95% | Terminated States: 4 | Alive States: 1
2018-01-31 20:07:54,495: [116141] m.ethereum:INFO: Starting symbolic transaction: 2
2018-01-31 20:07:54,554: [116141] m.ethereum:INFO: Finished symbolic transaction: 2 | Code Coverage: 95% | Terminated States: 4 | Alive States: 1
2018-01-31 20:07:54,565: [116141] m.ethereum:INFO: Generated testcase No. 4 - Account does not exist 1bfa530d5d685155e98cd7d9dd23f7b6a801cfef
2018-01-31 20:07:56,143: [116141] m.ethereum:INFO: Results in 
```

**Comment by feliam** [comment]
I guess next tx after suicide should do a TXERROR as the target address is no more.

**Comment by mossberg** [comment]
alternatively, we terminate the state right there. why would we keep it alive? because there might be other contracts that are still alive in the evm world?

**Comment by feliam** [comment]
The blockchain is changed after a suicide. There is one less accessible account (funds locked if sent to it ). And there was some eth transfer. 
So that may enable behaivors not reachable otherwise.

Ex. EXTCODECOPY of the suicided address behaves different after the suicide().

Probably not a source of common known issues?

**Comment by mossberg** [comment]
resolved as of cf72c12fd6a35ff9e62408fa23e033b94f804ceb
